framework/src/actions/InitProblemAction.C

```diff
@@ -28,4 +28,9 @@ InitProblemAction::act()
     _problem->init();
   else
     mooseError("Problem doesn't exist in InitProblemAction!");
+
+  // check for external variable to initialize
+  auto external_problem_ptr = std::dynamic_pointer_cast<ExternalProblem>(_problem);
+  if (external_problem_ptr)
+    external_problem_ptr->initializeVariable();
```

Comment:
any chance you can move that call to initializeVariable inside the ExternalProblem init() ?

Reply:
then you wont need to define these new functions to the two test objects

Reply:
So the issue I would run into there is that the FEProblemBase object _problem has not been initialized yet. Is there another way I can get a pointer to the external problem object at that point?

Reply:
I think it's inited you can see it right above
```
    _problem->init();
```

Reply:
Yes, but you wanted it moved into ExternalProblem.C right?

Reply:
_problem here is your local problem, of type ExternalProblem

Reply:
I see what you're saying. If I call initializeVariable inside of the ExternalProblem constructor, then I won't be able to override it in any derived classes though right?

Reply:
Or I can probably put all the logic for initializing variables in the ExternalVariable class and just not try to override it.

Reply:
I guess I should clarify, it can be overridden in a derived class but the parent constructor is not going to call the overridden version.

Reply:
the _problem->init call is not the constructor
you just need to redefine ExternalProblem::init to include the logic you want to add here

Reply:
You can call down to the base `FEProblemBase::init` from within your derived `ExternalProblem::init` `override`

Reply:
but sounds like we're not going to end up going down this path anyways

Reply:
That's what I ended up doing. I was going to push it this morning but as you said I'm not sure that this is the right solution.